### PR TITLE
[scheduler] Add namespace, queue and job related metrics

### LINKED ISSUE #757: Feature Request: Scheduler missing some important metrics (e.g. namespace, queue and job)
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:

Volcano scheduler already has many [metrics](https://github.com/volcano-sh/volcano/blob/v0.3/pkg/scheduler/metrics/metrics.go) to help user monitor itself. But there also some important metrics are missing, for example:

- Namespace share, weight and weighted share
- Queue share and weight
- Queue deserved, allocated and request resources
- Job share

**What you expected to happen**:

Add missing metrics

**How to reproduce it (as minimally and precisely as possible)**:

N/A

**Anything else we need to know?**:

N/A

**Environment**:

- Volcano Version: v0.3
- Kubernetes version (use `kubectl version`): v1.14.6
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by xiaogaozi**
Fix #757

### CODE REVIEW

File: pkg/scheduler/metrics/job.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package metrics
```
Comment (hzxuzhonghu) [inline pkg/scheduler/metrics/job.go:1]: add boilerplate header

File: pkg/scheduler/metrics/job.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto" // auto-registry collectors in default registry
+)
+
+var (
+	jobShare = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: VolcanoNamespace,
+			Name:      "job_share",
+			Help:      "Share for one job",
```
Comment (hzxuzhonghu) [inline pkg/scheduler/metrics/job.go:13]: add more description

Comment (xiaogaozi) [reply pkg/scheduler/metrics/job.go:13]: How about "DRF share for one job"?

File: pkg/scheduler/plugins/drf/drf.go [hunk 3]
```diff
@@ -113,7 +114,7 @@ func (drf *drfPlugin) OnSessionOpen(ssn *framework.Session) {
 			}
 			// all task in job should have the same namespace with job
 			nsOpts.allocated.Add(attr.allocated)
-			drf.updateShare(nsOpts)
+			drf.updateNsShare(job.Namespace, nsOpts)
```
Comment (k82cn) [inline pkg/scheduler/plugins/drf/drf.go:117]: let's use full name of Namespace :)

File: pkg/scheduler/metrics/job.go (hunk 2 above)
Comment (k82cn) [reply pkg/scheduler/metrics/job.go:13]: IMO, that should be enough for the first version :)

We should not highlight plugin in description, as it dependent on the configuration.

File: pkg/scheduler/metrics/job.go (hunk 1 above)
Comment (k82cn) [reply pkg/scheduler/metrics/job.go:1]: +1

**Comment by k82cn** [comment]
LGTM overall for this PR.

Ror the metrics, it's better to have a design doc on it as different users may have different requirements :)

File: pkg/scheduler/metrics/job.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (k82cn) [inline pkg/scheduler/metrics/job.go:2]: s/Kubernetes/Volcano

File: pkg/scheduler/metrics/namespace.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (k82cn) [inline pkg/scheduler/metrics/namespace.go:2]: ditto :)

File: pkg/scheduler/metrics/queue.go [hunk 6]
```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (k82cn) [inline pkg/scheduler/metrics/queue.go:2]: ditto :)

**Pull request merged**